### PR TITLE
Remove Middleware that was overwriting `starcounter-include.html`,

### PR DESCRIPTION
**Pull request by tomalec**
as it fails when file is cached.

Step to solve https://github.com/Starcounter/HeadsOmni/issues/272
~~Requires this ViewKeeper bug workaround to be moved to the custom element, https://github.com/Starcounter/starcounter-include/pull/116~~ (we decided on [alternative solution](https://github.com/Starcounter/Blending/pull/295#issuecomment-426255859))

### CODE REVIEW

**Comment by tomalec** [comment]
Instead of hard-coding the `<composition-inserter>` (BlendingEditor thing) inside `MasterWrapper.html` (BlendingProvider thing), I tried to achieve same result with blending :)

But it seems I'm doing something wrong, @warpech could you take a look at https://github.com/Starcounter/Blending/compare/HeadsOmni/issues/272-fix-providing-compositioninserter...HeadsOmni/issues/272-blend-together

Also, I'm not 100% sure that this is a better solution, as it requires 1 more request and loooots of more operations to just import single custom element.

**Comment by warpech** [comment]
In the branch https://github.com/Starcounter/Blending/tree/HeadsOmni/issues/272-fix-providing-compositioninserter-warp, I have pushed my fix for BlendingProvider that allows to have a consistent name of a namespace (`BlendingProviderComposition_0`), without changes in `starcounter-include` (https://github.com/Starcounter/starcounter-include/pull/116)

@tomalec please merge that branch into here if your agree

**Comment by tomalec** [comment]
```
StarcounterEnvironment.RunWithinApplication(Program.AppNameForJsonMerger, () =>
```
😱
Isn't it cheating? As we are pretending to be another app, plus it's deprecated
```
Obsolete("RunWithinApplication is no longer available. It breaks the isolation and independency between applications.")
``` 

----

:+1: For putting `starcounter-include` overwrites in `/src/BlendingProvider/wwwroot/BlendingProvider/views/MasterWrapper.html`. But if we can finally agree on one static value, we can as well put it in `starcounter-include` as well, not to sank in the sea of backward compatibility YAGNI

**Comment by warpech** [comment]
If we move the current messy detection of BlendingProvider into `starcounter-include`, we are shooting ourselves in the foot when it comes to maintainability. Basically any change to BlendingProvider in regards of namespacing would require simultaneous change to `starcounter-include`, and these two repos have a totally independent release cycle.

To use a deprecated API is a lesser evil that buys us time to create a more consistent fix.

**Comment by tomalec** [comment]
Merged @warpech's solution https://github.com/Starcounter/Blending/pull/295#issuecomment-426255859

**Comment by tomalec** [comment]
We have two problems here:

1. `body>starcounter-include` already tried and failed to find composition under old namespace, at the time when the `<script>changeScIncludesCompositionProvider()</script>` is executed. What's more it also sets `partialId` to null so the composition editor will ignore it.
2.  `BlendingProviderComposition_0` is hard-coded, so even if you change the app name, like `star --name="FooProvider"`, composition provider names space will still be `BlendingProviderComposition_0`.

I have a simple fix for each., but not for both. @warpech Should we ignore one of those cases?

I cannot find anything that solve both which is not another hack, like generating `MasterWrapper.html` dynamically in C#.

File: src/BlendingProvider/wwwroot/BlendingProvider/views/MasterWrapper.html [hunk 1]
```diff
@@ -1,7 +1,19 @@
...
+    <link rel="import" href="/BlendingEditor/elements/compositioneditor-inserter/compositioneditor-inserter.html">
+    <compositioneditor-inserter></compositioneditor-inserter>
+    <script>
+        (function () {
+            const constructor = customElements.get('starcounter-include');
+            if (typeof constructor !== 'undefined') {
+                constructor.prototype.compositionProvider = 'BlendingProviderComposition_0';
+            }
+        })();
+    </script>
```
[the remark this answers is not in the document]
Comment (warpech) [reply src/BlendingProvider/wwwroot/BlendingProvider/views/MasterWrapper.html]: As you can see in the `declarative-shadow-dom` part of this file, there is pretty much no reason to customize the composition of `MasterPage.html`. The point of BlendingProvider is to customize apps through layouts. This page is the container of a layout. It should be "headless" from the UI perspective.

**Comment by warpech** [comment]
Re 1. My response in https://github.com/Starcounter/Blending/pull/295#discussion_r222469845
Re 2. I like the solution from https://github.com/Starcounter/Blending/pull/295/commits/a25eb3cf1e34a2985b40d06e13fcb0633f23306b

**Comment by tomalec** [comment]
Thanks for support @warpech, 
I merged the branch with blended composition inserter, so there is no hard-coded cross-app dependency.

**Comment by warpech** [comment]
The reason why the test project did not start on TC was that you only applied the `blend.json` workaround to 2 out of 3 helper apps.

I added it to the remaining helper app. Now many of the tests pass.

**Comment by tomalec** [comment]
Fixed remaining flaky tests, and we are green to go!
I created separate issue to improve tests https://github.com/Starcounter/Blending/issues/306

Merging as the code is already reviewed.

**Pull request merged**